tests/nvmeof/workflows/ha.py

```diff
@@ -446,6 +460,127 @@ def power_on_off(self, gateway, action, wait_for_active_state=None):
+                )
+
+            LOG.info(
+                f"[ {hostname} ] Total time taken to scale down - {end_counter - start_counter} seconds"
+            )
+            return {
+                "scale-down-start-time": start_time,
+                "scale-down-end-time": end_time,
+                "scale-down-start-counter-time": start_counter,
+                "scale-down-end-counter-time": end_counter,
+            }
+
+    def scale_up(self, gateway_nodes):
+        """Scaling up of the NVMeoF Gateways.
+
+        Initiate scale-up
+        - Spin up the new gateways.
+        - Validate the ANA states of new GWs are optimized.
+
+        Post scale-up Validation
+        - List out namespaces associated with the new Gateways using ANA group ids.
+        - Check for 5 Consecutive times for the increments in write/read to validate IO continuation.
+        """
+        new_gws = []
+        start_counter = float()
+        start_time = str()
+        end_counter = float()
+        end_time = str()
+        gwnodes_to_be_deployed = list(set(self.config["gw_nodes"] + gateway_nodes))
+        self.config["gw_nodes"] = gwnodes_to_be_deployed
+        LOG.info(f"{gateway_nodes}: Scaling up NVMe Service")
+
+        # Scale up
+        deploy_nvme_service(self.cluster, self.config)
+        for gateway_node in gateway_nodes:
+            gw = get_node_by_id(self.cluster, gateway_node)
+            new_gws.append(NVMeGateway(gw, self.mtls))
+
+        for gateway in new_gws:
+            hostname = gateway.hostname
+
+            # Wait until 60 seconds
+            for w in WaitUntil():
+                # Check for gateway availability
+                if self.check_gateway_availability(gateway.ana_group_id):
+                    LOG.info(f"[ {gateway} ] NVMeofGW service is AVAILABLE.")
+                    state = self.get_optimized_state(gateway.ana_group_id)
+
+                    # check gateway for its own original path.
+                    if gateway.ana_group["name"] in state[0]:
+
+                        end_counter, end_time = get_current_timestamp()
+                        LOG.info(
+                            f"{hostname} restored to original path - {log_json_dump(state)}"
+                        )
+                        break
+
+                LOG.warning(f"[ {hostname} ] is still not in AVAILABLE state..")
+
+            if w.expired:
+                raise TimeoutError(
+                    f"[ {hostname} ] Scale up of NVMeofGW service failed after 60s timeout.."
+                )
+
+            LOG.info(
+                f"[ {hostname} ] Total time taken to scale up - {end_counter - start_counter} seconds"
+            )
+            return {
+                "scale-up-start-time": start_time,
+                "scale-up-end-time": end_time,
+                "scale-up-start-counter-time": start_counter,
+                "scale-up-end-counter-time": end_counter,
+            }
+
```

Comment:
Post scale up , we need to do below

1. Add listeners to newly added GWs
2. Create few namespaces so new namespaces will take ANA_GRP of new GWs
3. Disconnect-all and discover again and run IO from initiator for all namespaces especially keep eye on IO progression for newly added namespaces in logs

Reply:
I will take up this in next PR